iac-common/src/main/java/org/sonar/iac/common/reports/AbstractJsonReportImporter.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.common.reports;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.iac.common.warnings.AnalysisWarningsWrapper;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONArray;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONObject;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.JSONParser;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.ParseException;
+
+public abstract class AbstractJsonReportImporter {
+  private static final Logger LOG = Loggers.get(AbstractJsonReportImporter.class);
+  private static final JSONParser jsonParser = new JSONParser();
+
+  protected final SensorContext context;
+  private final AnalysisWarningsWrapper analysisWarnings;
+  private final String warningPrefix;
+  private Set<String> unresolvedPaths;
+
+  protected AbstractJsonReportImporter(SensorContext context, AnalysisWarningsWrapper analysisWarnings, String warningPrefix) {
```

Comment:
I'm not sure about this, as i don't have that much experience with our naming convention.
Shouldn't we drop the `Abstract` in the class name?

Reply:
We don't have a naming convention for that and for me, it's ok to use `Abstract` in the class name.